cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/io/cdap/cdap/etl/batch/mapreduce/MapReducePreparer.java

```diff
@@ -188,6 +189,32 @@ protected SubmitterPlugin createAggregator(BatchAggregator<?, ?, ?> aggregator,
     });
   }
 
+  @Override
+  protected SubmitterPlugin createReduceAggregator(BatchReduceAggregator<?, ?, ?> aggregator, StageSpec stageSpec) {
```

Comment:
is this any different than createAggregator? can they be combined somehow? Or maybe the closure be made into a single class that gets re-used?

Reply:
This is slightly different because the two aggregators are different types. 
To make the closure into a single class, the `TypeChecker` will have to change the signature to `getGroupKeyClass(Object aggregator)` and then check the type of the aggregator there. I feel it is better to keep the util class self explanatory from the method name.